### PR TITLE
Firebird support

### PR DESCRIPTION
**Pull request by turtuga**
Database driver for ibase/firebird

what works
- select 
- insert 
- update
- delete
- prepared statements
- create blob from file
- insert blog in database

I will try keep this updated

### CODE REVIEW

**Comment by jim-parry** [comment]
Your PR description is ... not helpful.
Also, we are not looking for Firebird contributions yet. 
Please wait until we "release" the CI4 repo, moving it to bcit-ci.

**Comment by turtuga** [comment]
Ok sorry

**Comment by jim-parry** [comment]
You're too eager :) We appreciate the intended contribution, but need to manage the integration of new components carefully! Please hold on to your changes for a bit longer :-/

We can keep the PR open for now, but you will need to fix your PR description... it is not indicative of what you are proposing to contribute to the framework.

**Comment by turtuga** [comment]
Create blob

```
$ficheiro = file_get_contents($_FILES['ficheiro']['tmp_name']);
        
$dados = $db->create_blob($ficheiro);
        
$db->query("INSERT INTO FICHEIROS (FICHEIRO, NOME) VALUES (?, ?)", [$dados, $_FILES['ficheiro']['name']]);
```

Num Rows
`$dados = $db->query("SELECT * FROM COUNTRY")->numRows();`

Result array
`$db->query("SELECT * FROM FICHEIROS")->getResultArray();`

Get Row
`$db->query("SELECT * FROM FICHEIROS")->getRow();`

File: system/Database/Ibase/Connection.php [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+ */
+
+use CodeIgniter\Database\BaseConnection;
+use CodeIgniter\Database\ConnectionInterface;
+use CodeIgniter\DatabaseException;
+
+/**
+ * Connection for Ibase
+ */
+class Connection extends BaseConnection implements ConnectionInterface
```
Comment (jim-parry) [inline system/Database/Ibase/Connection.php]: BaseConnection alrerady implements ConnectionInterface. It is redundant here.
I see the same thing in the other drivers, that you probably used as an example. 
I will fix them.

**Comment by jim-parry** [comment]
It's good that you are still wotking on this :)
A heads up ... you want to start signing off your commits, and move to GPG-signing soon. We are likely to start insisting on that.

**Pull request closed without merging**